vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1651,6 +1666,10 @@ def execute_model(
             assert model_output_broadcast_data is not None
             logits = model_output_broadcast_data["logits"]
 
+        # Apply batch state updates required for sampling stage;
+        # can run in parallel with model forward-pass.
+        self._update_sampling_states(scheduler_output)
```

Comment:
@njhill Doesn't this cause CPU-GPU synchronization?

Reply:
@WoosukKwon I don't think so, only non-blocking CPU->GPU of tensors that aren't used until the sampler...

Sync would only be for GPU->CPU right?